### PR TITLE
[feature] view::trim

### PR DESCRIPTION
**Pull request by h-2**

### CODE REVIEW

**Comment by h-2** [comment]
@sarahet This contains some things you will want to use for your PR. I ended putting all things into one class which seems best (and also the friend operator| prevents namespace problems).

File: include/seqan3/range/view/trim.hpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+ *
+ * \details
+ *
+ * This view can be used to do easy quality based trimming of sequences
+ *
+ * \par View properties
+ *
+ * |                                 |                                                                             |
+ * |---------------------------------|-----------------------------------------------------------------------------|
+ * | concepts required of `irng_t`   | `seqan3::input_range_concept<irng_t> && seqan3::quality_concept<value_type_t<irng_t>>>` |
```
Comment (sarahet) [inline include/seqan3/range/view/trim.hpp]: ```seqan3::quality_concept<value_type_t<irng_t>>>``` has one ```>``` too much.

Comment (h-2) [inline include/seqan3/range/view/trim.hpp]: good catch!

**Comment by h-2** [comment]
DONT MERGE YET:
  - [ ] update the view properties table to be like #103 
  - [ ] rename `view_trim` to `trim_fn` like ranges-v3 (`foo_fn` is the type of `view::foo` while `view_foo` is the type of the range returned by `view::foo(input_range)`)

**Pull request merged**